pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1962,8 +1964,12 @@ def compute_decomposition(wires: WiresLike) -> list[qml.operation.Operator]:
         ]
 
     def pow(self, z: Union[int, float]) -> list[qml.operation.Operator]:
-        z_mod4 = z % 4
-        return [ISWAP(wires=self.wires)] if z_mod4 == 2 else super().pow(z_mod4)
+        z_mod8 = z % 8
+        if z_mod8 == 2:
+            return [ISWAP(wires=self.wires)]
+        if z_mod8 == 4:
+            return [qml.Z(wires=self.wires[0]), qml.Z(wires=self.wires[1])]
+        return super().pow(z_mod8)
```

Comment:
```suggestion
        if abs(z_mod4 - 2) < 1e-6:
            return [ISWAP(wires=self.wires)]
        if abs(z_mod4 - 4) < 1e-6:
            return [qml.Z(wires=self.wires[0]), qml.Z(wires=self.wires[1])]
        return super().pow(z_mod8)
```
Same as above on other cases of `pow`, unless we actually don't care about other cases..